src/status_im/ui/screens/chat/views.cljs

```diff
@@ -390,13 +374,10 @@
                                           :row                message
                                           :idx                idx
                                           :list-ref           messages-list-ref}])
+           :initialNumToRender        initial-limit
```

Comment:
0

Reply:
I disagree with this one, it looks much better if the first 5-7 messages are loaded already.

Reply:
no it will delay opening of the screen on slow devices and design said it's a no no

Reply:
they said better have a blank screen than a lag

Reply:
if 5 messages are causing a lag, there is something wrong with message rendering and not with the container, tbh.

I think with you suggestion we just swipe the real issue under the rug and that's it.

Also, this we can also do later, if we won't be able to make rendering faster.

Reply:
yes there is an issue with the rendering, see https://discuss.status.im/t/performances-hunting-inefficiency/1250/8
but that means that for now, 0 is better, you can add a todo/warning that it has to be changed once we fix the bigger issue with the chat components

in my PR here https://github.com/status-im/status-react/pull/8364 I still didn't get them to render faster, the only thing I've left to try on cljs side is to pass the style as well so it's not computed in the component. After that it means we have to investigate lower levels, maybe reagent level of creating components, maybe it should even be done before idk

Reply:
well they render a little bit faster but it is still painfully slow on slow devices

Reply:
I still disagree that 0 is better for now, I think it is better to leave it as-is at the moment. When we will need to release 0.14.0, this code will be slightly faster than before, so that is an improvement to the quality of life of our users already.

If we make it 0, we will introduce blinking every time you open the chatroom, which is annoying. leaving this blinking is a last resort and I'm not convinced that we did try every other way than fixing that.

Reply:
no it will improve the situation, currently unless you have a small screen device you see blinking anyway with 5 messages, because it doesn't fill the screen, with 0 you will see the `chat-intro-header-container` and then enough messages to fill the screen will be displayed as soon as loaded. So that's an improvement compared to now where it just looks like you have a chat with 5 messages on high end devices, because you see the header and 5 last messages, and then suddenly more messages appear.

Reply:
I mean lets just ask @errorists what is the least worst situation here:
- show 5 messages and chat intro then fill messages inbetween later
- show chat header only then show messages when they are loaded

Reply:
it isn't 5 anymore, btw, but yeah, the size should be somehow calculated...
again, the major issue there is that our rendering of chat messages is slow for no reason. if all the possible 1.5 screen messages are loaded in < 0.3 second, you won't have this issue.

Reply:
@yenda also, you are talking like it is something that is already decided, and that we won't improve upon it. In the end there should be neither blinking nor delays to load < 50 text messages.

Reply:
6 is worst since the screen doesn't navigate until it's rendered so it's just lag from the user of low-end device perspective, and it is still not enough in most cases for the high devices to not show the header on half the screen

Reply:
No I'm talking like I already wanted to do what you are doing so I had this conversation and answer from @errorists was `better, open the blank screen as fast as possible, start rendering messages`

So my point is for this iteration, according to design, better render 0 for now. Then as soon as we figure how to render these messages as quick as they should we increase, potentially to the default flat list value to fill the screen and even a bit more

Reply:
okay, fine, I'm not really convinced, but I want this change to be merged asap :-P